src/App.tsx

```diff
@@ -1,23 +1,92 @@
 import "./App.css";
-
+import React, { useState } from 'react';
 import { taskList } from "./taskList";
 
+let taskListCopy = taskList;
+
+// Create a list with additional field to mark deleted task
+  let taskListExt: any[] = [];
+  for (let i=0 ; i < taskListCopy.length; i++){
```

Comment:
Having a `for` loop outside this in a small app is fine but this is not a good practice in general
The reason is that we want the list to be discarded (and thus, garbage collected) by the compiler at the end of the component life cycle instead of leaving it dangling when we don't use it anymore

My suggestion is to use `taskListCopy.map` with the `useState` default value directly